src/utils/deck.model.js

```diff
@@ -0,0 +1,48 @@
+const crypto = require('crypto');
+const Card = require('./card.model');
+
+class Deck {
+  constructor() {
+    const suits = ['Hearts', 'Diamonds', 'Clubs', 'Spades'];
+    this.cards = [];
+    suits.forEach((suit) => {
+      for (let value = 0; value < 15; value += 1) {
+        this.cards.push(new Card(suit, value));
+      }
+    });
+  }
+
+  shuffle() {
+    let currentIndex = this.cards.length;
+
+    while (currentIndex !== 0) {
+      const randomBuffer = new Uint32Array(1);
+      crypto.getRandomValues(randomBuffer);
+      const randomNumber = randomBuffer[0] / (0xffffffff + 1);
+      const randomIndex = Math.floor(randomNumber * (60));
+      currentIndex -= 1;
+
+      [this.cards[currentIndex], this.cards[randomIndex]] = [
+        this.cards[randomIndex], this.cards[currentIndex]];
+    }
+  }
+
+  deal(round, playerNumber) {
+    const hands = [];
+    for (let i = 0; i < playerNumber; i += 1) {
+      hands.push([]);
+      for (let j = 1; j <= round; j += 1) {
```

Comment:
There is still a SonarCloud issue on this line, but it's up to you if you want to fix it since this is a (odd) minor issue and the expected behavior of the loop. It could probably be fixed by using `Array.from` or something similar.

Reply:
You could also use a `// NOSONAR not used in secure contexts` comment to not show this error in sonarcloud

Reply:
added the comment to the line